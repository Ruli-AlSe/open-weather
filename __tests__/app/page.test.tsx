import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';

import Page from '@/app/page';
import { JSX } from 'react';

async function resolvedComponent(Component: () => Promise<JSX.Element>) {
  const ComponentResolved = await Component();
  return () => ComponentResolved;
}

describe('Main Page', () => {
  it('renders a heading', async () => {
    const ResolvedPage = await resolvedComponent(Page);
    render(<ResolvedPage />);

    const heading = screen.getByRole('heading', { level: 1 });

    expect(heading).toBeInTheDocument();
    expect(heading).toHaveTextContent('Weather Forecast');
  });
});
