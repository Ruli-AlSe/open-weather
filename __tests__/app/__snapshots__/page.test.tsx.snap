// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Page matches snapshot 1`] = `
<DocumentFragment>
  <main
    aria-label="Weather forecast application"
    class="w-full flex flex-col items-center min-h-screen p-4 sm:p-10 bg-black"
    role="main"
  >
    <h1
      aria-label="Weather Forecast"
      class="flex flex-col text-center sm:flex-row gap-3 text-5xl tracking-wider mt-16"
    >
      <span
        aria-hidden="true"
      >
        üåç
      </span>
       
      <span
        class="bg-gradient-to-r from-orange-500 to-orange-300 text-transparent bg-clip-text"
      >
        <b>
          Weather
        </b>
         Forecast
      </span>
    </h1>
    <section
      aria-label="Main weather information"
      class="flex flex-col-reverse md:grid md:grid-cols-3"
      role="region"
    >
      <div
        class="col-span-2 flex flex-col"
      >
        <section
          aria-label="City search"
          class="md:px-3 mt-10 flex justify-center gap-3"
          id="search-bar-section"
        >
          <div
            class="w-full relative"
          >
            <h2
              class="text-2xl font-bold my-4"
            >
              Search for any city in the world and click on it
            </h2>
            <div
              aria-controls="city-list"
              aria-expanded="false"
              aria-haspopup="listbox"
              role="combobox"
            >
              <input
                aria-autocomplete="list"
                aria-controls="city-list"
                aria-label="Search for a city"
                class="w-full p-2 md:p-3 text-xl rounded md:rounded-lg text-black"
                placeholder="Enter location"
                type="text"
              />
            </div>
          </div>
        </section>
      </div>
      <section
        aria-label="Favorite cities list"
        class="px-5"
        role="region"
      >
        <h2
          class="text-2xl font-bold my-4"
        >
          Favorite Cities
        </h2>
        <div
          aria-label="List of favorite cities"
          class="px-1 grid grid-cols-1 gap-4 text-white max-h-60 md:max-h-[26rem] overflow-y-scroll"
          id="favorite-cities"
          role="list"
        >
          <p
            role="alert"
          >
            No favorite cities added yet.
          </p>
        </div>
      </section>
    </section>
    <section
      aria-label="Detailed weather forecasts"
      role="region"
    />
  </main>
</DocumentFragment>
`;
